src/inmanta/protocol/common.py

```diff
@@ -505,14 +505,14 @@ def _validate_return_type(self, arg_type: Type) -> None:
         if typing_inspect.is_generic_type(arg_type) and issubclass(typing_inspect.get_origin(arg_type), ReturnValue):
             self._validate_type_arg(arg, typing_inspect.get_args(arg_type, evaluate=True)[0], allow_none_type=True)
 
-        elif not typing_inspect.is_generic_type(arg_type) and issubclass(arg_type, ReturnValue):
-            raise InvalidMethodDefinition("ReturnValue should have a type specified.")
-
-        elif not typing_inspect.is_generic_type(arg_type) and issubclass(arg_type, type(None)):
-            pass
+        elif not typing_inspect.is_generic_type(arg_type) and isinstance(arg_type, type):
+            if issubclass(arg_type, ReturnValue):
+                raise InvalidMethodDefinition("ReturnValue should have a type specified.")
 
+            elif issubclass(arg_type, ReturnValue):
```

Comment:
Why is this here? This is exactly the same as the if condition

Reply:
Good catch, this should be issubclass(arg_type, type(None)). I accidentally changed it.

Reply:
Though actually this is redundant with `allow_none_type=True`, I'm gonna leave it out.